f/connectors/alerts/alerts_gcs.py

```diff
@@ -124,221 +131,251 @@ def _get_tif_rel_filepath(blob_name, local_file_path, territory_id):
-            )
-
-            # if jpeg file already exist skip it
-            if os.path.exists(os.path.join(rel_filepath, jpeg_file)):
-                continue
-
-            logger.info(f"Converting TIFF file to JPEG: {jpeg_file}")
-
-            # Ensure the file exists in the local directory
-            if os.path.exists(os.path.join(rel_filepath, file_name.split("/")[-1])):
-                try:
-                    with Image.open(
-                        os.path.join(rel_filepath, file_name.split("/")[-1])
-                    ) as img:
-                        # save the image at the same location in the datalake as its tiff
-                        img.save(os.path.join(rel_filepath, jpeg_file), "JPEG")
-                except Exception as e:
-                    logger.error(
-                        f"Tif image can not be opened, potentially empty: {str(e)}"
-                    )
-        elif file_name.endswith(".csv"):
-            if file_name != alerts_metadata_filename:
-                continue
-            file_path = os.path.join(path, file_name)
-            if os.path.exists(file_path):
-                df = pd.read_csv(file_path)
-                filtered_df = df[df["territory_id"] == territory_id]
-                # Hash the file content
-                hash_hex = sha256sum(file_path)
-                # Generate a UUID from the content hash
-                metadata_uuid = uuid.uuid5(uuid.NAMESPACE_DNS, hash_hex)
-
-                filtered_df["_geolocation"] = [[0.0, 0.0]] * len(filtered_df)
-                filtered_df["metadata_uuid"] = [str(metadata_uuid)] * len(filtered_df)
-                filtered_df["source"] = file_name
-                # this op is used for terras alerts only at this point
-                filtered_df["alert_source"] = "terras"
-                alerts_metadata = filtered_df.to_dict("records")
-        else:
-            logger.info(f"Skipping file: {file_name}")
-
-    return {"geojsons": geojsons, "alerts_metadata": alerts_metadata}
-
-
-def load_alerts_to_postgis(local_directory, processed):
-    outputs = {}
-    outputs["alerts_metadata"] = processed["alerts_metadata"]
-    outputs["geojsons"] = []
-
-    for file_name in processed["geojsons"]:
-        if os.path.exists(os.path.join(local_directory, file_name)):
+    # Convert CSV bytes to DataFrame
+    df = pd.read_csv(StringIO(alerts_metadata))
+
+    # Filter DataFrame based on territory_id
+    filtered_df = df[df["territory_id"] == territory_id].copy()
+
+    # Generate a unique UUID for each row
+    filtered_df["metadata_uuid"] = filtered_df.apply(_generate_uuid, axis=1)
+
+    # TODO: Currently, this script is only used for Terras alerts. Let's discuss a more sustainable approach with the alerts provider(s).
+    filtered_df.loc[:, "alert_source"] = "terras"
+
+    # Replace all NaN values with None
+    filtered_df = filtered_df.replace({nan: None})
```

Comment:
Is there a way to do this without `numpy.nan`?

Reply:
```suggestion
    filtered_df = filtered_df.fillna(None)
```

Reply:
This does not seem to work: "ValueError: Must specify a fill 'value' or 'method'."

For posterity, I also tried `filtered_df = filtered_df.where(pd.notna(filtered_df), None)` which did not seem to work.

Reply:
you are right.  If your goal is to not `import numpy`, you can do this:

`filtered_df.replace({float("nan"): None})`


`nan` is how pandas represents missing data, and it's efficient in a column whose datatype is float.  Replacing  `nan` to `None` will upgrade that column's dtype to `object`.  In this case it's not a problem, but won't always be what you want.